### PR TITLE
iface_hotplug: Added time delay to take effect after detachment

### PR DESCRIPTION
**Pull request by MalleshKoti**
Signed-off-by: Mallesh Koti <mallesh@linux.vnet.ibm.com>

### CODE REVIEW

File: libvirt/tests/src/virtual_network/iface_hotplug.py [hunk 1]
```diff
@@ -218,11 +219,17 @@ def create_iface_xml(mac):
...
+                        polltime = time.time() + timeout
+                        while True:
+                            # Check interface in dumpxml output
+                            if not vm_xml.VMXML.get_iface_by_mac(vm_name,
+                                                             iface['mac']):
+                                break
+                            else:
+                                time.sleep(2)
+                                if time.time() > polltime:
+                                    raise error.TestFail("Interface still "
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_network/iface_hotplug.py]: please use `test.fail()` here and elsewhere

File: libvirt/tests/src/virtual_network/iface_hotplug.py [hunk 2]
```diff
@@ -218,11 +219,17 @@ def create_iface_xml(mac):
...
+                                                 " still after detachment")
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_network/iface_hotplug.py]: please fix the indentation

File: libvirt/tests/src/virtual_network/iface_hotplug.py [hunk 3]
```diff
@@ -218,11 +219,17 @@ def create_iface_xml(mac):
...
-                        # Check interface in dumpxml output
-                        if vm_xml.VMXML.get_iface_by_mac(vm_name,
-                                                         iface['mac']):
-                            raise error.TestFail("Interface still exist"
-                                                 " after detachment")
+                        polltime = time.time() + timeout
+                        while True:
+                            # Check interface in dumpxml output
+                            if not vm_xml.VMXML.get_iface_by_mac(vm_name,
+                                                             iface['mac']):
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_network/iface_hotplug.py]: same here

File: libvirt/tests/cfg/virtual_network/iface_hotplug.cfg [hunk 4]
```diff
@@ -3,6 +3,7 @@
     take_regular_screendumps = "no"
     start_vm = "yes"
     status_error = "no"
+    timeout = 10
```
Comment (balamuruhans) [inline libvirt/tests/cfg/virtual_network/iface_hotplug.cfg]: can be `poll_timeout = 10`

**Comment by MalleshKoti** [comment]
Have made the required changes. Thanks

File: libvirt/tests/src/virtual_network/iface_hotplug.py [hunk 5]
```diff
@@ -218,11 +219,17 @@ def create_iface_xml(mac):
...
+                        while True:
+                            # Check interface in dumpxml output
+                            if not vm_xml.VMXML.get_iface_by_mac(vm_name,
+                                                                 iface['mac']):
+                                break
+                            else:
+                                time.sleep(2)
+                                if time.time() > polltime:
+                                    raise test.fail("Interface still "
+                                                    " still after detachment")
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_network/iface_hotplug.py]: 1. "Interface still exists after detachment" can be used as there are 2 still in the error message.
2. `raise` is not required

**Comment by MalleshKoti** [comment]
Thanks Balamuruhans and chunfuwen.

**Pull request merged**